include/common/mir/immediate_executor.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © 2022 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIR_IMMEDIATE_EXECUTOR_H_
+#define MIR_IMMEDIATE_EXECUTOR_H_
+
+#include "executor.h"
+
+namespace mir
+{
+/// An Executor that runs work on the current thread within spawn()
+extern Executor& immediate_executor;
```

Comment:
Should we have a
```
class ImmediateExecutor : public Executor { };

extern ImmediateExecutor& immediate_executor;
```

We already have `NonBlockingExecutor` for using the type to describe the requirement on the executor; since `immediate_executor` is going to be used for cases where the non-deferred-ness is necessary for correctness, maybe we should be able to express that, too?

Reply:
+1

Reply:
I fail to see how a visible ImmediateExecutor type would be useful (aside from maybe using dynamic_cast<>ing to check if things are it, which we probably want to avoid). Since there's only one possible implementation of ImmediateExecutor, anything that would take an ImmediateExecutor could simply take nothing and hard-code in the one to use.

Reply:
That's a reasonable argument, but is it actually the case that there's only one possible implementation of `ImmediateExecutor`? I'm not committed either way, but it seems a `MockImmediateExecutor` would be possibly useful, or *maybe* an implementation that caught and logged all exceptions or something?